### PR TITLE
add blue cell light info for network status

### PR DESCRIPTION
**Pull request by MCTBL**
add new [blue cell light](https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/pull/574) counter to network status
G@7ZPY`PI4%IC5{6N}O1_QI
and remove repeat lang entry

### CODE REVIEW

**Comment by Dream-Master** [comment]
@chochem the blue light was merged so i like to add this as well

**Comment by MCTBL** [comment]
yesterday forget to add this

File: src/main/java/appeng/api/implementations/tiles/IChestOrDrive.java [hunk 1]
```diff
@@ -31,6 +31,8 @@ public interface IChestOrDrive extends ICellContainer, IGridHost, IOrientable {
      * <p>
      * 1 - green,
      * <p>
+     * 2 - blue,
+     * <p>
      * 2 - orange,
      * <p>
      * 3 - red
```
Comment (OneEyeMaker) [inline src/main/java/appeng/api/implementations/tiles/IChestOrDrive.java:38]: Please, fix other indices too.

File: src/main/java/appeng/client/gui/implementations/GuiNetworkStatus.java [hunk 2]
```diff
@@ -430,14 +430,18 @@ private void drawItemInfo() {
                 16,
                 GuiColors.DefaultBlack.getColor());
         this.fontRendererObj.drawString(
-                GuiText.Green.getLocal() + " : "
+                GuiText.Green.getLocal().charAt(0) + " : "
```
Comment (OneEyeMaker) [inline src/main/java/appeng/client/gui/implementations/GuiNetworkStatus.java:433]: * First character of color name isn't clear *enough* as a label.
* In some locales first character can even cause confusion.

Please find a nice way to display all color names without such abbreviation. And apply such rule in all similar cases below.

**Review by OneEyeMaker — CHANGES_REQUESTED** [review]

**Comment by MCTBL** [comment]
now it looks like
image

**Comment by MCTBL** [comment]
https://github.com/user-attachments/assets/08786e68-fb7f-4755-8493-cf5d79b87dc8

KCVB{GR}KFR(F}2}{NYA}P1
ZH4R5(3}RV(FIW4GAR_VT6I
NMI14~8I%NY`0D{QWM4XQ1U

File: src/main/java/appeng/me/cache/GridStorageCache.java [hunk 3]
```diff
@@ -84,14 +88,20 @@ public class GridStorageCache implements IStorageGrid {
...
     private long essentiaCellO;
     private long essentiaCellR;
     private long essentiaCellCount;
     private int ticksCount;
     private int networkBytesUpdateFrequency;
-    private static final int CELL_RED = 3;
-    private static final int CELL_ORANGE = 2;
+    private final HashMap<ItemStack, Integer> itemCells = new HashMap<>();
+    private final HashMap<ItemStack, Integer> fluidCells = new HashMap<>();
+    private final HashMap<ItemStack, Integer> essentiaCells = new HashMap<>();
```
Comment (OneEyeMaker) [inline src/main/java/appeng/me/cache/GridStorageCache.java:99]: `ItemStack` type is not designed for *hash-based* containers (leke `HashMap` and `HashSet`). So I suggest to use specific container - `ItemStackMap`. Please use implementation of `ItemStackMap` *provided by GTNHLib*.

File: src/main/java/appeng/me/cache/GridStorageCache.java [hunk 4]
```diff
@@ -485,19 +513,45 @@ private void updateBytesInfo() {
...
+
+    private void putEssentiaItemStackIntoMap(final HashMap<ItemStack, Integer> map, final ItemStack stack) {
+        ItemStack newStack = new ItemStack(stack.getItem());
+        newStack.setItemDamage(stack.getItemDamage());
+        if (map.containsKey(newStack)) {
+            map.put(newStack, map.get(newStack) + 1);
+        } else {
+            map.put(newStack, 1);
+        }
+    }
```
Comment (OneEyeMaker) [inline src/main/java/appeng/me/cache/GridStorageCache.java:546]: These methods can be combined into one. Also code can be simplified:
```suggestion
    private void putItemStackIntoMap(final ItemStackMap<Integer> map, final ItemStack stack) {
        ItemStack newStack = new ItemStack(stack.getItem(), 1, stack.getItemDamage());
        if (map.computeIfPresent(newStack, (currentStack, count) -> count + 1) == null) {
            map.put(newStack, 1);
        }
    }
```

File: src/main/java/appeng/me/cache/GridStorageCache.java [hunk 5]
```diff
@@ -514,9 +569,26 @@ private void resetCellInfo() {
...
+        return itemCells;
+    }
+
+    public HashMap<ItemStack, Integer> getFluidCells() {
+        return fluidCells;
+    }
+
+    public HashMap<ItemStack, Integer> getEssentiaCells() {
+        return essentiaCells;
     }
```
Comment (OneEyeMaker) [inline src/main/java/appeng/me/cache/GridStorageCache.java:592]: Change return type of these methods to `ItemStackMap<Integer>`.

File: src/main/java/appeng/container/implementations/ContainerNetworkStatus.java [hunk 6]
```diff
@@ -167,39 +181,64 @@ public void detectAndSendChanges() {
...
+                    final GridStorageCache sg = this.network.getCache(IStorageGrid.class);
+                    CellType selectedCellType = AEConfig.instance.selectedCellType();
+                    HashMap<ItemStack, Integer> cells = null;
+                    if (selectedCellType == CellType.ITEM) {
+                        cells = sg.getItemCells();
+                    } else if (selectedCellType == CellType.FLUID) {
+                        cells = sg.getFluidCells();
+                    } else if (selectedCellType == CellType.ESSENTIA) {
+                        cells = sg.getEssentiaCells();
+                    }
```
Comment (OneEyeMaker) [inline src/main/java/appeng/container/implementations/ContainerNetworkStatus.java:213]: Can be simplified using modern `switch` syntax:
```suggestion
                    ItemStackMap<Integer> cells = switch (selectedCellType) {
                        case ITEM -> sg.getItemCells();
                        case FLUID -> sg.getFluidCells();
                        case ESSENTIA -> sg.getEssentiaCells();
                    };
```

**Review by OneEyeMaker — CHANGES_REQUESTED** [review]

File: src/main/java/appeng/me/cache/GridStorageCache.java (hunk 3 above)
Comment (MCTBL) [reply src/main/java/appeng/me/cache/GridStorageCache.java:99]: but that will make ae **need** GTNH to run. or just use stack name be the key?

File: src/main/java/appeng/me/cache/GridStorageCache.java (hunk 4 above)
Comment (MCTBL) [reply src/main/java/appeng/me/cache/GridStorageCache.java:546]: thats cool, thx

File: src/main/java/appeng/me/cache/GridStorageCache.java [hunk 7]
```diff
@@ -376,114 +388,112 @@ public void applyChanges() {
...
+            case CELL_BLUE -> essentiaCellB++;
+            case CELL_ORANGE -> essentiaCellO++;
+            case CELL_RED -> essentiaCellR++;
+        }
+        this.essentiaTypesTotal += typesTotalToAdd;
+        this.essentiaTypesUsed += typesUsedToAdd;
+        this.essentiaCellCount++;
+    }
+
+    private void updateCellsStatusUseICCRAndStack(final ICellCacheRegistry iccr, final ItemStack newCellStack) {
```
Comment (OneEyeMaker) [inline src/main/java/appeng/me/cache/GridStorageCache.java:436]:
```suggestion
    private void updateCellsStatusFromRegistry(final ICellCacheRegistry iccr, final ItemStack newCellStack) {
```

File: src/main/java/appeng/me/cache/GridStorageCache.java [hunk 8]
```diff
@@ -492,12 +502,20 @@ private void updateBytesInfo() {
         }
     }
 
+    private void putItemStackIntoMap(final ItemStackMap<Integer> map, final ItemStack stack, final int cellStatus) {
+        ItemStack newStack = new ItemStack(stack.getItem(), 1, stack.getItemDamage());
+        if (map.computeIfPresent(newStack, (currentStack, stackCount) -> stackCount += 1) == null) {
```
Comment (OneEyeMaker) [inline src/main/java/appeng/me/cache/GridStorageCache.java:507]: No need in assignment - you just need to return new value from this function.
```suggestion
        if (map.computeIfPresent(newStack, (currentStack, stackCount) -> stackCount + 1) == null) {
```

**Review by OneEyeMaker — CHANGES_REQUESTED** [review]

**Review by OneEyeMaker — APPROVED** [review]

**Pull request merged**